here-naksha-lib-core/src/test/java/com/here/naksha/lib/core/util/diff/PatcherTest.java

```diff
@@ -57,6 +56,48 @@ void basic() throws IOException {
     assertNull(newDiff);
   }
 
+  @Test
+  void testBasicNestedJson() {
+    final JsonObject f3 =
+            JsonSerializable.deserialize(IoHelp.readResource("patcher/feature_3.json"), JsonObject.class);
+    assertNotNull(f3);
+    final JsonObject f4 =
+            JsonSerializable.deserialize(IoHelp.readResource("patcher/feature_4.json"), JsonObject.class);
+    assertNotNull(f4);
+
+    final Difference diff34 = Patcher.getDifference(f3, f4);
+    assertNotNull(diff34);
+    assert (diff34 instanceof MapDiff);
+    final MapDiff mapDiff34 = (MapDiff) diff34;
+    assertTrue(mapDiff34.get("isAdded") instanceof InsertOp);
+    assertTrue(mapDiff34.get("map") instanceof MapDiff);
+    assertTrue(mapDiff34.get("array") instanceof ListDiff);
+    final MapDiff nestedMapDiff34 = (MapDiff) mapDiff34.get("map");
+    assertTrue(nestedMapDiff34.get("willBeUpdated") instanceof UpdateOp);
+    assertTrue(nestedMapDiff34.get("willBeDeleted") instanceof RemoveOp);
+    final ListDiff nestedArrayDiff34 = (ListDiff) mapDiff34.get("array");
+    assertTrue(nestedArrayDiff34.get(0) instanceof MapDiff);
+    assertTrue(((MapDiff) nestedArrayDiff34.get(0)).get("isAddedProperty") instanceof InsertOp);
+    assertTrue(((MapDiff) nestedArrayDiff34.get(0)).get("willBeDeletedProperty") instanceof RemoveOp);
+    assertTrue(nestedArrayDiff34.get(1) instanceof RemoveOp);
+
+    final JsonObject f5 =
```

Comment:
I also suggest splitting big test into multiple ones... for example F3 and F4 is separate than F3 and F5 comparison